robottelo/host_helpers/satellite_mixins.py

```diff
@@ -324,3 +325,8 @@ def api_factory(self):
         if not getattr(self, '_api_factory', None):
             self._api_factory = APIFactory(self)
         return self._api_factory
+
+    def ui_factory(self, session):
+        if not getattr(self, '_ui_factory', None):
+            self._ui_factory = UIFactory(session=session)
```

Comment:
```suggestion
            self._ui_factory = UIFactory(self)
```
if you have self.ui attribute in Satellite class